src/main/java/uk/gov/pay/api/service/PaymentSearchService.java

```diff
@@ -0,0 +1,172 @@
+                                    paymentUriGenerator.getPaymentURI(baseUrl, charge.getChargeId()),
+                                    paymentUriGenerator.getPaymentEventsURI(baseUrl, charge.getChargeId()),
+                                    paymentUriGenerator.getPaymentCancelURI(baseUrl, charge.getChargeId()),
+                                    paymentUriGenerator.getPaymentRefundsURI(baseUrl, charge.getChargeId())))
+                            .collect(Collectors.toList());
+                    HalRepresentation.HalRepresentationBuilder halRepresentation = HalRepresentation.builder()
+                            .addProperty("results", chargeFromResponses)
+                            .addProperty("count", searchResponse.getCount())
+                            .addProperty("total", searchResponse.getTotal())
+                            .addProperty("page", searchResponse.getPage());
+                    addLink(halRepresentation, "self", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getSelf()));
+                    addLink(halRepresentation, "first_page", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getFirstPage()));
+                    addLink(halRepresentation, "last_page", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getLastPage()));
+                    addLink(halRepresentation, "prev_page", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getPrevPage()));
+                    addLink(halRepresentation, "next_page", transformIntoPublicUri(baseUrl, searchResponse.getLinks().getNextPage()));
+                
+                    return halRepresentation.build();
+                } catch (IOException | ProcessingException | URISyntaxException ex) {
+                    throw new SearchChargesException(ex);
+                }
+            }
+            throw new SearchChargesException(connectorResponse);
+        }
+        throw new NotImplementedException("DD serch is not yet implemented");
+    }
+
+    private String getConnectorUrl(Account account, List<Pair<String, String>> queryParams) {
+        String urlPath = connectorUriGenerator.chargesURI(account);
+        UriBuilder builder = UriBuilder
+                .fromPath(urlPath);
+        queryParams.stream().forEach(pair -> {
+            if (isNotBlank(pair.getRight())) {
+                builder.queryParam(pair.getKey(), pair.getValue());
+            }
+        });
+        return builder.toString();
+    }
+
+    private boolean isDirectDebitAccount(Account account) {
+        return account.getPaymentType().equals(DIRECT_DEBIT);
+    }
+
+    private void addLink(HalRepresentation.HalRepresentationBuilder halRepresentationBuilder, String name, URI uri) {
+        if (uri != null) {
+            halRepresentationBuilder.addLink(name, uri); 
+        }
+    }
+
+    private URI transformIntoPublicUri(String baseUrl, uk.gov.pay.api.model.links.Link link) throws URISyntaxException {
+        if (link == null)
```

Comment:
Is there any case when this value can be null? Also, what happens if we return null?

Reply:
I don't know, this code is mainly copy paste from `PaymentsResource`. I assume there is a reason why it is how it is.